docs/DeveloperGuide.md

```diff
@@ -2,36 +2,530 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+1. Reference to AB-3 Developer Guide
+
+- [Source](https://se-education.org/addressbook-level3/DeveloperGuide.html#proposed-undoredo-feature)
+- Used as template to structure this Developer Guide
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+<img src="UML Diagrams/Architecture_Diagram.png" width="280">
+
+The architecture diagram given above explains the high level design of the application. The diagram depicts the key
+component of the application that enables it to provide its functionalities.
+
+Majority of the app's work is done by the following components
```

Comment:
As Joshua said, maybe it's better to have storage component and try to have more components